src/MPAGSCipher/ProcessCommandLine.cpp

```diff
@@ -55,6 +55,25 @@ bool processCommandLine(const std::vector<std::string>& cmdLineArgs,
                 settings.outputFile = cmdLineArgs[i + 1];
                 ++i;
             }
+        } else if (cmdLineArgs[i] == "--multi-cipher") {
+            // Check its not the last argument
+            if (i == nCmdLineArgs - 1) {
+                std::cerr << "[error] --multi-cipher requires a positive integer argument"
+                        << std::endl;
+                processStatus = false;
+                break;
+            } else {
+                std::istringstream iss(cmdLineArgs[i + 1]);
+                if (!(iss >> settings.nExpectedCiphers) ||
+                    settings.nExpectedCiphers <= 0) {
+                    std::cerr
+                        << "[error] --multi-cipher requires a positive integer argument"
+                        << std::endl;
+                    processStatus = false;
+                    break;
+                }
+                i++;
```

Comment:
This is a slightly different approach from ours but should work equally well I think.  It's very good that you're considering the various possible failure conditions.